shesha-core/src/Shesha.Application/StoredFiles/StoredFileController.cs

```diff
@@ -59,13 +56,43 @@ public class StoredFileController : ControllerBase, ITransientDependency
         [HttpGet, Route("Download")]
         public async Task<FileStreamResult> Download(Guid id, int? versionNo)
         {
-            return await GetShaFileStreamResultAsync(id, isView: false, versionNo);
+            var fileVersion = await GetStoredFileVersionAsync(id, versionNo);
+            var fileContents = await _fileService.GetStreamAsync(fileVersion);
+            await _fileService.MarkDownloadedAsync(fileVersion);
+
+            return File(fileContents, fileVersion.FileType.GetContentType(), fileVersion.FileName);
+        }
+
+        [HttpGet, Route("Base64String")]
+        public async Task<IActionResult> GetBase64StringAsync(Guid id, int? versionNo)
+        {
+            var fileVersion = await GetStoredFileVersionAsync(id, versionNo);
+            var fileContents = await _fileService.GetStreamAsync(fileVersion);
```

Comment:
@azarielmochusi A memory leak here. `fileContents` is a stream and you don't dispose it